pkgs/development/libraries/wayland/scanner.nix

```diff
@@ -0,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  wayland,
+  meson,
+  pkg-config,
+  ninja,
+  wayland-scanner,
+  expat,
+  libxml2,
+}:
+
+stdenv.mkDerivation {
+  pname = "wayland-scanner";
+  inherit (wayland) version src;
+
+  outputs = [
+    "out"
+    "bin"
+    "dev"
+  ];
+  separateDebugInfo = true;
+
+  mesonFlags = [
+    "-Ddocumentation=false"
+    "-Dlibraries=false"
+    "-Dtests=false"
+  ];
+
+  depsBuildBuild = [ pkg-config ];
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    ninja
+  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) wayland-scanner;
```

Comment:
I'm tempted to test if this is really still necessary (how come we can build native wayland-scanner without a prebuilt wayland-scanner, but the cross wayland-scanner requires a prebuilt native wayland-scanner)

Nit:

<details>

`optionals cond [ x ]` over `optional cond x` because this prevents the accidental `optional cond [ x ]`

</details>

Reply:
It is necessary, as it fails to otherwise fails to find `wayland-scanner` for host:

```
Run-time dependency expat found: YES 2.6.2
Run-time dependency libxml-2.0 found: YES 2.13.3
Program embed.py found: YES (/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3.12 /build/wayland-1.23.0/src/embed.py)
Found pkg-config: YES (/nix/store/blb1ci2nwrmgzb3g1xl4mvqpwkhm3s3b-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Did not find CMake 'cmake'
Found CMake: NO
Build-time dependency wayland-scanner found: NO (tried pkgconfig)

src/meson.build:84:15: ERROR: Dependency "wayland-scanner" not found, tried pkgconfig

A full log can be found at /build/wayland-1.23.0/build/meson-logs/meson-log.txt
error: build of '/nix/store/v16pnil9wkd1hhvrfp7n6sbgl86000ms-wayland-scanner-aarch64-unknown-linux-gnu-1.23.0.drv' on 'ssh://root@ws.my.flokli.io' failed: builder for '/nix/store/v16pnil9wkd1hhvrfp7n6sbgl86000ms-wayland-scanner-aarch64-unknown-linux-gnu-1.23.0.drv' failed with exit code 1
```

Reply:
I personally ignore that nit, if we want that it should be a tree-wide thing, because why otherwise do we have it.

Reply:
😱. Thanks.